cmake/EthCompilerSettings.cmake

```diff
@@ -22,6 +22,13 @@ if(CCACHE_FOUND)
 	message("Using ccache")
 endif(CCACHE_FOUND)
 
+include(CheckCXXCompilerFlag)
+
+check_cxx_compiler_flag(-fstack-protector-strong stack_protector_support)
+if (stack_protector_support)
+	add_compile_options(-fstack-protector-strong)
```

Comment:
More / less yes. The `-strong` one is stronger. The strongest is `-all`. I think this only selects what function to protect...

We are skipping an exotic case where a compiler supports `-fstack-protector` but not `-fstack-protector-strong`. I can leave with this. This does not affect solidity releases.